### PR TITLE
remove func GetStrategy()

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

func GetStrategy() is deprecated, use Strategy() instead.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/ok-to-test

Code is no longer referenced in openshift/builder, this is safe to merge with a subsequent minor version bump on the next release.

File: pkg/build/strategies/strategies.go [hunk 1]
```diff
@@ -13,18 +13,13 @@ import (
...
-// GetStrategy decides what build strategy will be used for the STI build.
-// TODO: deprecated, use Strategy() instead
-func GetStrategy(client docker.Client, config *api.Config) (build.Builder, api.BuildInfo, error) {
-	return Strategy(client, config, build.Overrides{})
-}
-
 // Strategy creates the appropriate build strategy for the provided config, using
 // the overrides provided. Not all strategies support all overrides.
-func Strategy(client docker.Client, config *api.Config, overrides build.Overrides) (build.Builder, api.BuildInfo, error) {
+func Strategy(client docker.Client, config *api.Config) (build.Builder, api.BuildInfo, error) {
```
Comment (adambkaplan) [inline pkg/build/strategies/strategies.go]: Please revert - this will break openshift/builder

File: pkg/build/strategies/strategies.go [hunk 2]
```diff
@@ -13,18 +13,13 @@ import (
...
 	var builder build.Builder
 	var buildInfo api.BuildInfo
 	var err error
+	var overrides build.Overrides
```
Comment (adambkaplan) [inline pkg/build/strategies/strategies.go]: Ditto

File: pkg/cmd/cli/cmd/build.go [hunk 3]
```diff
@@ -162,7 +162,7 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 
 			glog.V(2).Infof("\n%s\n", describe.Config(client, cfg))
 
-			builder, _, err := strategies.GetStrategy(client, cfg)
+			builder, _, err := strategies.Strategy(client, cfg)
```
Comment (adambkaplan) [inline pkg/cmd/cli/cmd/build.go]: Pass an empty `build.Overrides` for all invocations

File: test/integration/docker/integration_test.go [hunk 4]
```diff
@@ -298,7 +298,7 @@ func (i *integrationTest) exerciseCleanAllowedUIDsBuild(tag, imageName string, e
 		ExcludeRegExp:     tar.DefaultExclusionPattern.String(),
 	}
 	config.AllowedUIDs.Set("1-")
-	_, _, err := strategies.GetStrategy(engineClient, config)
+	_, _, err := strategies.Strategy(engineClient, config)
```
Comment (adambkaplan) [inline test/integration/docker/integration_test.go]: Pass empty `build.Overrides`

File: pkg/build/strategies/strategies.go (hunk 1 above)
Comment (soulseen) [reply pkg/build/strategies/strategies.go]: How about define an empty build.Overrides{} in this function?

Comment (adambkaplan) [reply pkg/build/strategies/strategies.go]: This function is referenced by another project - changing signature breaks the API.

**Pull request closed without merging**